### PR TITLE
mockgcp: initial support for StorageBucket

### PR DESCRIPTION
**Pull request by justinsb**
A little unusual because we don't have the v1 service proto (though we
have the request/response definitions)

### CODE REVIEW

**Comment by justinsb** [comment]
Oh actually this one is sort of tricky because we're also generating the proto from the OpenAPI (though this is all only for mocks at the end of the day, so low-consequence).  So I think it would be good if @cheftako could also take a look.

/assign @cheftako

File: mockgcp/mockstorage/v1.go [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+	// for _, path := range paths {
+	// 	switch path {
+	// 	case "description":
+	// 		obj.Description = req.GetCertificate().GetDescription()
+	// 	case "labels":
+	// 		obj.Labels = req.GetCertificate().GetLabels()
+	// 	default:
+	// 		return nil, status.Errorf(codes.InvalidArgument, "update_mask path %q not valid", path)
+	// 	}
+	// }
```
Comment (gemmahou) [inline mockgcp/mockstorage/v1.go:104]: nit: remove commented lines

Comment (justinsb) [reply mockgcp/mockstorage/v1.go]: Good catch - done (and there were more later on also!)

**Comment by justinsb** [comment]
Rebased to pick up test fixes

**Comment by gemmahou** [comment]
This PR need a rebase, and refactor the error handling for CRUD functions.

**Comment by justinsb** [comment]
Thanks @gemmahou  ... rebased and cleaned up!

**Pull request merged**